lib/amigo/router.rb

```diff
@@ -11,7 +11,13 @@ class Router
     def perform(event_json)
       event_name = event_json["name"]
       matches = Amigo.registered_event_jobs.
-        select { |job| File.fnmatch(job.pattern, event_name, File::FNM_EXTGLOB) }
+        select do |job|
+        if job.pattern.instance_of?(Regexp)
```

Comment:
```ruby
if job.pattern.respond_to?(:match)
   job.pattern.match(event_name)
else
  File.fnmatch(job.pattern, event_name, File::FNM_EXTGLOB)
end
```

Reply:
@rgalanakis `.responds_to?(:match)` also matches strings, so this isn't a good solution to check if it's a regular expression. Got any other ideas?

Reply:
Ah, ok. Use `is_a?(Regexp)`

https://stackoverflow.com/questions/3893278/ruby-kind-of-vs-instance-of-vs-is-a

Reply:
@rgalanakis Great, done. Now I know the differences of those methods. 👍🏼